### ISSUE TITLE
Cannot build Centos or Stretch Docker images.

### ISSUE THREAD

**Issue by ecc256** [openresty/docker-openresty#92]
Cannot build Centos Docker image.
bionic and alpine builds work fine.

Centos build error:
Bad value for --with-lua: /usr/local/openresty/luajit is not a valid directory.

Full build log is below:
```
docker build -f ~/Dockerfile  -t openresty-centos .

Sending build context to Docker daemon  8.923MB
Step 1/25 : ARG RESTY_IMAGE_BASE="centos"
Step 2/25 : ARG RESTY_IMAGE_TAG="7"
Step 3/25 : FROM ${RESTY_IMAGE_BASE}:${RESTY_IMAGE_TAG}
7: Pulling from library/centos
8ba884070f61: Pull complete 
Digest: sha256:8d487d68857f5bc9595793279b33d082b03713341ddec91054382641d14db861
Status: Downloaded newer image for centos:7
 ---> 9f38484d220f
Step 4/25 : LABEL maintainer="Evan Wies <evan@neomantra.net>"
 ---> Running in 441e786b571c
Removing intermediate container 441e786b571c
 ---> 330741172740
Step 5/25 : ARG RESTY_IMAGE_BASE="centos"
 ---> Running in d18cd0ee00fa
Removing intermediate container d18cd0ee00fa
 ---> 22402c808f51
Step 6/25 : ARG RESTY_LUAROCKS_VERSION="3.0.4"
 ---> Running in cfbf9814a90c
Removing intermediate container cfbf9814a90c
 ---> 0ee1b0fdc4bf
Step 7/25 : ARG RESTY_RPM_FLAVOR=""
 ---> Running in 31a7af64a5f0
Removing intermediate container 31a7af64a5f0
 ---> 2b54a17ee416
Step 8/25 : ARG RESTY_RPM_VERSION="1.15.8.1rc1-0.el7.centos"
 ---> Running in 3e9995feff00
Removing intermediate container 3e9995feff00
 ---> 989da8f8cacc
Step 9/25 : ARG RESTY_RPM_ARCH="x86_64"
 ---> Running in 7b53ca127e2e
Removing intermediate container 7b53ca127e2e
 ---> dd762503c6e1
Step 10/25 : LABEL resty_luarocks_version="${RESTY_LUAROCKS_VERSION}"
 ---> Running in 310759266aa2
Removing intermediate container 310759266aa2
 ---> b0b643c97cd8
Step 11/25 : LABEL resty_rpm_flavor="${RESTY_RPM_FLAVOR}"
 ---> Running in a878cf18856a
Removing intermediate container a878cf18856a
 ---> 4235533105b6
Step 12/25 : LABEL resty_rpm_version="${RESTY_RPM_VERSION}"
 ---> Running in 14a3cee7edd6
Removing intermediate container 14a3cee7edd6
 ---> 4b1ca8169f1d
Step 13/25 : LABEL resty_rpm_arch="${RESTY_RPM_ARCH}"
 ---> Running in b0b5f0310268
Removing intermediate container b0b5f0310268
 ---> 264f3db05666
Step 14/25 : RUN yum-config-manager --add-repo https://openresty.org/package/${RESTY_IMAGE_BASE}/openresty.repo     && yum install -y         gettext         make         openresty${RESTY_RPM_FLAVOR}-${RESTY_RPM_VERSION}.${RESTY_RPM_ARCH}         openresty-opm-${RESTY_RPM_VERSION}         openresty-resty-${RESTY_RPM_VERSION}         unzip     && cd /tmp     && curl -fSL https://github.com/luarocks/luarocks/archive/${RESTY_LUAROCKS_VERSION}.tar.gz -o luarocks-${RESTY_LUAROCKS_VERSION}.tar.gz     && tar xzf luarocks-${RESTY_LUAROCKS_VERSION}.tar.gz     && cd luarocks-${RESTY_LUAROCKS_VERSION}     && ./configure         --prefix=/usr/local/openresty/luajit         --with-lua=/usr/local/openresty/luajit         --lua-suffix=jit-2.1.0-beta3         --with-lua-include=/usr/local/openresty/luajit/include/luajit-2.1     && make build     && make install     && cd /tmp     && rm -rf luarocks-${RESTY_LUAROCKS_VERSION} luarocks-${RESTY_LUAROCKS_VERSION}.tar.gz     && yum remove -y make     && yum clean all     && ln -sf /dev/stdout /usr/local/openresty/nginx/logs/access.log     && ln -sf /dev/stderr /usr/local/openresty/nginx/logs/error.log
 ---> Running in d9bc89047bdc
Loaded plugins: fastestmirror, ovl
adding repo from: https://openresty.org/package/centos/openresty.repo
grabbing file https://openresty.org/package/centos/openresty.repo to /etc/yum.repos.d/openresty.repo
repo saved to /etc/yum.repos.d/openresty.repo
Loaded plugins: fastestmirror, ovl
Determining fastest mirrors
 * base: mirror.cogentco.com
 * extras: mirror.cogentco.com
 * updates: mirror.cogentco.com
No package openresty-1.15.8.1rc1-0.el7.centos.x86_64 available.
No package openresty-opm-1.15.8.1rc1-0.el7.centos available.
No package openresty-resty-1.15.8.1rc1-0.el7.centos available.
Resolving Dependencies
--> Running transaction check
---> Package gettext.x86_64 0:0.19.8.1-2.el7 will be installed
--> Processing Dependency: gettext-libs(x86-64) = 0.19.8.1-2.el7 for package: gettext-0.19.8.1-2.el7.x86_64
--> Processing Dependency: libgomp.so.1(GOMP_1.0)(64bit) for package: gettext-0.19.8.1-2.el7.x86_64
--> Processing Dependency: libunistring.so.0()(64bit) for package: gettext-0.19.8.1-2.el7.x86_64
--> Processing Dependency: libgomp.so.1()(64bit) for package: gettext-0.19.8.1-2.el7.x86_64
--> Processing Dependency: libgettextsrc-0.19.8.1.so()(64bit) for package: gettext-0.19.8.1-2.el7.x86_64
--> Processing Dependency: libgettextlib-0.19.8.1.so()(64bit) for package: gettext-0.19.8.1-2.el7.x86_64
--> Processing Dependency: libcroco-0.6.so.3()(64bit) for package: gettext-0.19.8.1-2.el7.x86_64
---> Package make.x86_64 1:3.82-23.el7 will be installed
---> Package unzip.x86_64 0:6.0-19.el7 will be installed
--> Running transaction check
---> Package gettext-libs.x86_64 0:0.19.8.1-2.el7 will be installed
---> Package libcroco.x86_64 0:0.6.12-4.el7 will be installed
---> Package libgomp.x86_64 0:4.8.5-36.el7 will be installed
---> Package libunistring.x86_64 0:0.9.3-9.el7 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package              Arch           Version                 Repository    Size
================================================================================
Installing:
 gettext              x86_64         0.19.8.1-2.el7          base         1.0 M
 make                 x86_64         1:3.82-23.el7           base         420 k
 unzip                x86_64         6.0-19.el7              base         170 k
Installing for dependencies:
 gettext-libs         x86_64         0.19.8.1-2.el7          base         501 k
 libcroco             x86_64         0.6.12-4.el7            base         105 k
 libgomp              x86_64         4.8.5-36.el7            base         157 k
 libunistring         x86_64         0.9.3-9.el7             base         293 k

Transaction Summary
================================================================================
Install  3 Packages (+4 Dependent packages)

Total download size: 2.6 M
Installed size: 9.3 M
Downloading packages:
warning: /var/cache/yum/x86_64/7/base/packages/gettext-libs-0.19.8.1-2.el7.x86_64.rpm: Header V3 RSA/SHA256 Signature, key ID f4a80eb5: NOKEY
Public key for gettext-libs-0.19.8.1-2.el7.x86_64.rpm is not installed
--------------------------------------------------------------------------------
Total                                               13 MB/s | 2.6 MB  00:00     
Retrieving key from file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
Importing GPG key 0xF4A80EB5:
 Userid     : "CentOS-7 Key (CentOS 7 Official Signing Key) <security@centos.org>"
 Fingerprint: 6341 ab27 53d7 8a78 a7c2 7bb1 24c6 a8a7 f4a8 0eb5
 Package    : centos-release-7-6.1810.2.el7.centos.x86_64 (@CentOS)
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : libunistring-0.9.3-9.el7.x86_64                              1/7 
  Installing : libgomp-4.8.5-36.el7.x86_64                                  2/7 
  Installing : libcroco-0.6.12-4.el7.x86_64                                 3/7 
  Installing : gettext-libs-0.19.8.1-2.el7.x86_64                           4/7 
  Installing : gettext-0.19.8.1-2.el7.x86_64                                5/7 
install-info: No such file or directory for /usr/share/info/gettext.info.gz
  Installing : unzip-6.0-19.el7.x86_64                                      6/7 
  Installing : 1:make-3.82-23.el7.x86_64                                    7/7 
  Verifying  : libcroco-0.6.12-4.el7.x86_64                                 1/7 
  Verifying  : libgomp-4.8.5-36.el7.x86_64                                  2/7 
  Verifying  : gettext-0.19.8.1-2.el7.x86_64                                3/7 
  Verifying  : libunistring-0.9.3-9.el7.x86_64                              4/7 
  Verifying  : 1:make-3.82-23.el7.x86_64                                    5/7 
  Verifying  : unzip-6.0-19.el7.x86_64                                      6/7 
  Verifying  : gettext-libs-0.19.8.1-2.el7.x86_64                           7/7 

Installed:
  gettext.x86_64 0:0.19.8.1-2.el7           make.x86_64 1:3.82-23.el7          
  unzip.x86_64 0:6.0-19.el7                

Dependency Installed:
  gettext-libs.x86_64 0:0.19.8.1-2.el7     libcroco.x86_64 0:0.6.12-4.el7       
  libgomp.x86_64 0:4.8.5-36.el7            libunistring.x86_64 0:0.9.3-9.el7    

Complete!
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   124    0   124    0     0    365      0 --:--:-- --:--:-- --:--:--   366
  0     0    0 5248k    0     0  4689k      0 --:--:--  0:00:01 --:--:-- 7173k
Bad value for --with-lua: /usr/local/openresty/luajit is not a valid directory.

configure failed.

The command '/bin/sh -c yum-config-manager --add-repo https://openresty.org/package/${RESTY_IMAGE_BASE}/openresty.repo     && yum install -y         gettext         make         openresty${RESTY_RPM_FLAVOR}-${RESTY_RPM_VERSION}.${RESTY_RPM_ARCH}         openresty-opm-${RESTY_RPM_VERSION}         openresty-resty-${RESTY_RPM_VERSION}         unzip     && cd /tmp     && curl -fSL https://github.com/luarocks/luarocks/archive/${RESTY_LUAROCKS_VERSION}.tar.gz -o luarocks-${RESTY_LUAROCKS_VERSION}.tar.gz     && tar xzf luarocks-${RESTY_LUAROCKS_VERSION}.tar.gz     && cd luarocks-${RESTY_LUAROCKS_VERSION}     && ./configure         --prefix=/usr/local/openresty/luajit         --with-lua=/usr/local/openresty/luajit         --lua-suffix=jit-2.1.0-beta3         --with-lua-include=/usr/local/openresty/luajit/include/luajit-2.1     && make build     && make install     && cd /tmp     && rm -rf luarocks-${RESTY_LUAROCKS_VERSION} luarocks-${RESTY_LUAROCKS_VERSION}.tar.gz     && yum remove -y make     && yum clean all     && ln -sf /dev/stdout /usr/local/openresty/nginx/logs/access.log     && ln -sf /dev/stderr /usr/local/openresty/nginx/logs/error.log' returned a non-zero code: 1
```

**Comment by neomantra** [comment]
In moving to the 1.15.x RC series, I had to disable `centos` and `stretch` builds and left their respective Dockerfiles in an "almost ready" state.  They are ready to grab their upstream packages (with some minor version number tuning) once those are released.

If you want to build your own images, check out a tag such as `1.13.6.2-2`.

Ideally I would have done this with a branch, but our Travis image building system doesn't deal with branches.  I opted to make `HEAD` not work on a those flavors for a few weeks.   Another option was to keep the image builds private and just push them to Docker Hub... but for the community, I prefer the public build logs that Travis provides.   So sorry for the hassle and thanks for the report!

Keeping this open until the upstream images are available and the builds all work.
